daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/Interface.scala

```diff
@@ -16,24 +16,6 @@ import scala.jdk.CollectionConverters._
 
 sealed abstract class InterfaceType extends Product with Serializable {
   def `type`: DefDataType.FWT
-
-  def fold[Z](normal: DefDataType.FWT => Z, template: (Record.FWT, DefTemplate[Type]) => Z): Z =
-    this match {
-      case InterfaceType.Normal(typ) => normal(typ)
-      case InterfaceType.Template(typ, tpl) => template(typ, tpl)
-    }
-
-  /** Alias for `type`. */
-  def getType: DefDataType.FWT = `type`
-  def getTemplate: j.Optional[_ <: DefTemplate.FWT] =
-    fold(
-      { _ =>
-        j.Optional.empty()
-      },
-      { (_, tpl) =>
-        j.Optional.of(tpl)
-      },
-    )
```

Comment:
The only usage of this was to check is an `InterfaceType` was a `Template`. I made a function for this closer to the usage site.

Reply:
These functions were here for better Java support for using the interface library, rather than our own use, like all the `get*` functions therein; we can decide not to do that anymore but probably separately.

Reply:
Makes sense, I actually left something analogous in another place for this reason.